pkg/pipelines/bootstrap.go

```diff
@@ -289,7 +290,7 @@ func bootstrapResources(o *BootstrapOptions, appFs afero.Fs) (res.Resources, err
 	}
 	bootstrapped[pipelinesFile] = m
 
-	k.Resources = append(k.Resources, secretFilename, imageRepoBindingFilename)
+	k.Resources = utility.RemoveDuplicates(append(k.Resources, secretFilename, imageRepoBindingFilename))
```

Comment:
I wonder if we shouldn't change this to be a bit nicer?

how about `k.AddResource(secretFilename, imageRepoBindingFilename)` ?

And delegate the task of avoiding duplicates to the Resources type?

This means that we don't need to call `RemoveDuplicates` everywhere?

Reply:
Perhaps something like https://github.com/redhat-developer/kam/tree/avoid-duplicate-resources ?

Reply:
@bigkevmcd I like your improvement with one concern.  I don't want to call the remove duplicate function every time `AddResources()` is called.   I should only need to remove any duplicate once.    Can you raise a PR to remove this one?  Thanks.

Reply:
@wtam2018 duplicates will only be removed once? `removeDuplicatesAndSort` only removes duplicates _if_ they exist?

Reply:
@bigkevmcd  Correct.  Only once after the last resource added to the slice.